x-pack/plugins/observability_shared/public/services/get_context_menu_items_from_actions.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiContextMenuItemProps } from '@elastic/eui';
+import { buildContextMenuForActions, UiActionsStart } from '@kbn/ui-actions-plugin/public';
+import { ObservabilityTriggerId } from '../../common';
+
+export function getContextMenuItemsFromActions({
+  uiActions,
+  triggerId,
+  context,
+}: {
+  uiActions: UiActionsStart;
+  triggerId: ObservabilityTriggerId;
+  context: Record<string, any>;
+}): Promise<EuiContextMenuItemProps[]> {
+  return uiActions
+    .getTriggerCompatibleActions(triggerId, context)
+    .then((actions) => {
+      return buildContextMenuForActions({
```

Comment:
This function returns panel descriptors, and using panels in the consuming apps would be too big of a change, so I'm just getting the items from the created panels (and assuming people don't build complex panels using these extension points :) ).